WalletWasabi/Wallets/Wallet.cs

```diff
@@ -621,7 +625,7 @@ private async Task<Block> DownloadBlockAsync(uint256 hash, CancellationToken can
 
 						// Select a random node we are connected to.
 						Node node = Nodes.ConnectedNodes.RandomElement();
-						if (node == default(Node) && !node.IsConnected && Synchronizer.Network == Network.RegTest)
+						if (node == default(Node) && !node.IsConnected && Network == Network.RegTest)
```

Comment:
`Node` is a reference type what means `default(Node)` is `null`. So, in case `node == default(Node)` then the next condition (`!node.IsConnected`) has to raise a NRE.

This same line of code is in the `TransactionBroadcaster`

Reply:
Nice catch. I think this isn't raising NRE usually, because the node is found, but this is raising the peer not connected stuff!